onnxruntime/core/providers/rocm/tunable/gemm_tunable.cuh

```diff
@@ -78,6 +79,80 @@ class GemmTunableOp : public tunable::TunableOp<GemmParams<T>> {
 #endif
 };
 
+template <typename T, typename ALayout, typename BLayout>
+class BatchedGemmTunableOp : public tunable::TunableOp<BatchedGemmParams<T>> {
+ public:
+  BatchedGemmTunableOp() {
+    this->ops_.emplace_back(RocBlasBatchedGemmOp<T>);
+  }
+
+  const BatchedGemmParams<T>* PreTuning(const BatchedGemmParams<T>* params) override {
+    if (!IsZero(params->beta)) {
+      // See GemmTunableOp<T>::PreTuning for more details
+      BatchedGemmParams<T>* proxy = new BatchedGemmParams<T>();
+      *proxy = *params;
+
+      // malloc a large buffer and then slice it
+      const int single_buffer_bytes = CeilDiv(proxy->m * proxy->ldc * sizeof(T), 128) * 128;
+      T* buffer;
+      HIP_CALL_THROW(hipMalloc(&buffer, proxy->batch * single_buffer_bytes));
+      std::vector<T*> buffer_ptrs(proxy->batch, nullptr);
+      for (int i = 0; i < proxy->batch; i++) {
+        // note the following is offseted by bytes
+        buffer_ptrs[i] = reinterpret_cast<T*>(reinterpret_cast<char*>(buffer) + i * single_buffer_bytes);
+      }
+
+      // copy all ptrs to device
+      HIP_CALL_THROW(hipMalloc(&(proxy->cs), proxy->batch * sizeof(T*)));
+      HIP_CALL_THROW(hipMemcpy(proxy->cs, buffer_ptrs.data(), buffer_ptrs.size() * sizeof(T*), hipMemcpyHostToDevice));
+      return proxy;
+    }
+
+    return params;
+  }
+
+  void PostTuning(const BatchedGemmParams<T>* params) override {
+    if (!IsZero(params->beta)) {
+      T* buffer;
+      HIP_CALL_THROW(hipMemcpy(&buffer, params->cs, sizeof(T*), hipMemcpyDeviceToHost));
+      HIP_CALL_THROW(hipFree(buffer));
+      HIP_CALL_THROW(hipFree(params->cs));
+      delete params;
+    }
+  }
+};
+
+template <typename T, typename ALayout, typename BLayout>
+class StridedBatchedGemmTunableOp : public tunable::TunableOp<StridedBatchedGemmParams<T>> {
+ public:
+  StridedBatchedGemmTunableOp() {
+    this->ops_.emplace_back(RocBlasStridedBatchedGemmOp<T>);
+    for (auto&& [_, op] : GetCKStridedBatchedGemmTypeStringAndOps<T, ALayout, BLayout>()) {
+      ORT_UNUSED_PARAMETER(_);
+      this->ops_.emplace_back(std::move(op));
+    }
+  }
+
+  const StridedBatchedGemmParams<T>* PreTuning(const StridedBatchedGemmParams<T>* params) override {
+    if (!IsZero(params->beta)) {
+      // See GemmTunableOp<T>::PreTuning for more details
+      StridedBatchedGemmParams<T>* proxy = new StridedBatchedGemmParams<T>();
+      *proxy = *params;
+      HIP_CALL_THROW(hipMalloc(&(proxy->c), proxy->m * proxy->stride_c * sizeof(T)));
```

Comment:
> `hipMalloc(&(proxy->c), proxy->m * proxy->stride_c * sizeof(T))` 

Shouldn't this be `hipMalloc(&(proxy->c), proxy->m * proxy->n * proxy->batch * sizeof(T))`. i.e., `proxy->c` should have size `m x n x batch_size` elems, whereas with your computation, `proxy->c` would have size ~`m x (m x n)` elems, because `proxy->stride_c = m x n x sizeof(T) + padding`, I'd assume?

Reply:
It is bugged here, this should be `batch * stride_c`.

But no, you must not use `proxy->m * proxy->n * proxy->batch` to compute. When the user call it with a stride, they mean it, so the padding is necessary, otherwise, you offset into an unknown wasted land.

Reply:
Fixed.